saleor/checkout/forms.py

```diff
@@ -1,18 +1,42 @@
 from django import forms
-from django.contrib.auth.models import AnonymousUser
 from django.utils.translation import ugettext_lazy as _
+from saleor.userprofile.models import Address
 
-from ..userprofile.forms import AddressForm
 
+class AddressChoiceField(forms.ModelChoiceField):
+    choice_values = []
 
-class ShippingForm(AddressForm):
+    def validate(self, value):
+        if value not in self.choice_values:
+            return super(AddressChoiceField, self).validate(value)
 
-    shipping_same_as_billing = forms.BooleanField(initial=True, required=False)
+    def to_python(self, value):
+        if value in self.choice_values:
+            return value
+        else:
+            return super(AddressChoiceField, self).to_python(value)
 
+    def add_magic_choices(self, magic_choices, queryset):
```

Comment:
This seems to cross layers of separation. Can't we pass all necessary information using `__init__` and `__iter__`?